### PR TITLE
Ajax: trigger error callback on native abort

### PR DESCRIPTION
**Pull request by timmywil**
Fixes gh-2079

### CODE REVIEW

File: test/unit/ajax.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ QUnit.module( "ajax", {
 	);
 
 	ajaxTest( "jQuery.ajax() - success callbacks", 8, function( assert ) {
-	    return {
+			return {
```
Comment (scottgonzalez) [inline test/unit/ajax.js]: Bad indentation.

Comment (timmywil) [inline test/unit/ajax.js]: Actually, this converts spaces to tabs.

Comment (mzgol) [inline test/unit/ajax.js]: Yeah, but ut;s still indented one tab too many, isn't it?

File: test/unit/ajax.js [hunk 2]
```diff
@@ -437,6 +437,26 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - native abort", 2, function( assert ) {
+		return {
+			url: url( "data/name.php?wait=1" ),
+			xhr: function() {
+				var xhr = new window.XMLHttpRequest();
+				setTimeout( function() {
+					xhr.statusText = "abort";
```
Comment (gibson042) [inline test/unit/ajax.js]: Why?

File: test/unit/ajax.js (hunk 1 above)
Comment (timmywil) [inline test/unit/ajax.js]: Good call.

File: test/unit/ajax.js (hunk 2 above)
Comment (timmywil) [inline test/unit/ajax.js]: Debugging, woops.

**Comment by timmywil** [comment]
Bah, IE9 doesn't have onabort. The only other solution I see is to use .onreadystatechange, but that would change other stuff as well.

**Comment by dmethvin** [comment]
Could we just document that IE9 doesn't support this? It would be a shame to let IE9 prevent us from doing the right thing for everyone else.

**Comment by timmywil** [comment]
That's probably better than switching everything back to `onreadystatechange`. Anyone against this?

**Comment by gibson042** [comment]
Looking at the compat code, I don't think use of `onreadystatechange` would be that cumbersome.

**Comment by timmywil** [comment]
Afaict, it does require determining success/error manually again, which has historically been cumbersome.

**Comment by timmywil** [comment]
Yea, we end up losing a lot of advantages.

**Comment by gibson042** [comment]
No, we can keep the current structure and only use `onreadystatechange` for the abort case.

**Comment by timmywil** [comment]
Ah, well I tried that first, but `onreadystatechange` is called before other handlers, and brings up some IE9 quirks, such as not being to access any properties on a manual, native abort – which makes it difficult to test – and not having a solid way to tell the difference between abort and other errors.

**Comment by gibson042** [comment]
For example, verify `readyState === 4` and then delay its body (inside of which is an `if ( callback )` guard) by a tick so that it only executes if none of `onload`/`onerror`/`onabort` have.

**Comment by timmywil** [comment]
Ah, a `setTimeout` could do the trick to allow other callbacks to run first.

**Comment by timmywil** [comment]
@dmethvin @gibson042 This works. Let me know if you have ideas to get it smaller.

**Comment by jaubourg** [comment]
> Ah, a `setTimeout` could do the trick to allow other callbacks to run first.

Except for synchronous requests, right?

**Comment by timmywil** [comment]
> Except for synchronous requests, right?

Right

**Pull request closed without merging**